pkg/cloudprovider/cloudprovider.go

```diff
@@ -100,6 +100,32 @@ func New(ctx awscontext.Context) *CloudProvider {
 	}
 }
 
+func (c *CloudProvider) GetDriftedNodes(ctx context.Context, provisioner *v1alpha5.Provisioner, nodes []v1.Node) []v1.Node {
+	driftedNodesMap := map[string]bool{}
+	var driftedNodes []v1.Node
+	//Consider if provisioner should be a part of this interface
+	for _, drifter := range []interface {
+		GetDriftedNodes([]v1.Node) []v1.Node
+	}{
+		&AmiDrifter{
+			provisioner: provisioner,
+			ctx:         ctx,
+			c:           c,
+		},
+	} {
+		//Execute Drifters
+		for _, node := range drifter.GetDriftedNodes(nodes) {
+			//Check first in the map, one node can be drifted by multiple drifter implementations
+			if _, ok := driftedNodesMap[node.Name]; !ok {
+				driftedNodes = append(driftedNodes, node)
+			}
+			driftedNodesMap[node.Name] = true
+		}
+	}
+	logging.FromContext(ctx).Infof("Found %d drifted nodes", len(driftedNodes))
```

Comment:
My hunch is that this will be unnecessarily noisy in logs. I don't think that we'll need to log the number of drifted nodes we find, since this could be called many times in rapid succession. I think it'd be better if we just log if we annotate any node as drifted in the controller.go file. WDYT?